File: java/app/src/main/java/com/cleversolutions/ads/android/sampleapp/SampleActivity.java
package com.cleversolutions.ads.android.sampleapp;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.cleversolutions.ads.AdCallback;
import com.cleversolutions.ads.AdError;
import com.cleversolutions.ads.AdLoadCallback;
import com.cleversolutions.ads.AdSize;
import com.cleversolutions.ads.AdStatusHandler;
import com.cleversolutions.ads.AdType;
import com.cleversolutions.ads.AdViewListener;
import com.cleversolutions.ads.LoadingManagerMode;
import com.cleversolutions.ads.MediationManager;
import com.cleversolutions.ads.android.CAS;
import com.cleversolutions.ads.android.CASBannerView;

import java.util.Objects;

public class SampleActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Validate Integration
        CAS.validateIntegration(this);

        // Get current SDK version
        ((TextView) findViewById(R.id.casVersionText)).setText(CAS.getSDKVersion());

        MediationManager adManager = Objects.requireNonNull(SampleApplication.adManager);
        createBanner(adManager);
        createInterstitial(adManager);
        createRewarded(adManager);
    }

    private void createBanner(MediationManager manager) {
        CASBannerView bannerView = new CASBannerView(this, manager);

        // Set required Ad size
        bannerView.setSize(AdSize.getAdaptiveBannerInScreen(this));
        //bannerView.setSize(AdSize.BANNER);
        //bannerView.setSize(AdSize.LEADERBOARD);
        //bannerView.setSize(AdSize.MEDIUM_RECTANGLE);

        TextView label = findViewById(R.id.bannerStatusText);
        // Set Ad content listener
        bannerView.setAdListener(new AdViewListener() {
            @Override
            public void onAdViewPresented(@NonNull CASBannerView casBannerView, @NonNull AdStatusHandler adStatusHandler) {
                label.setText("Presented: " + adStatusHandler.getNetwork());
                Log.d(SampleApplication.TAG, "Banner Ad presented from " + adStatusHandler.getNetwork());
            }

            @Override
            public void onAdViewLoaded(@NonNull CASBannerView casBannerView) {
                label.setText("Loaded");
                Log.d(SampleApplication.TAG, "Banner Ad loaded and ready to present");
            }

            @Override
            public void onAdViewFailed(@NonNull CASBannerView casBannerView, @NonNull AdError adError) {
                label.setText(adError.getMessage());
                Log.e(SampleApplication.TAG, "Banner Ad received error: " + adError.getMessage());
            }

            @Override
            public void onAdViewClicked(@NonNull CASBannerView casBannerView) {
                label.setText("Clicked");
                Log.d(SampleApplication.TAG, "Banner Ad received Click action");
            }
        });

        // Set controls
        findViewById(R.id.loadBannerBtn).setOnClickListener(v -> {
            label.setText("Loading");
            bannerView.loadNextAd();
        });

        findViewById(R.id.showBannerBtn).setOnClickListener(v -> {
            bannerView.setVisibility(View.VISIBLE);
        });

        findViewById(R.id.hideBannerBtn).setOnClickListener(v -> {
            bannerView.setVisibility(View.GONE);
        });

        // Add view to container
        LinearLayout container = findViewById(R.id.container);
        container.addView(bannerView);
    }

    private void createInterstitial(MediationManager manager) {
        TextView label = findViewById(R.id.interStatusText);

        // Set Ad load callback
        manager.getOnAdLoadEvent().add(new AdLoadCallback() {
            @Override
            public void onAdLoaded(@NonNull AdType adType) {
                if (adType == AdType.Interstitial) {
                    label.setText("Loaded");
                    Log.d(SampleApplication.TAG, "Interstitial Ad loaded and ready to show");
                }
            }

            @Override
            public void onAdFailedToLoad(@NonNull AdType adType, @Nullable String s) {
                if (adType == AdType.Interstitial) {
                    label.setText(s);
                    Log.d(SampleApplication.TAG, "Interstitial Ad received error: " + s);
                }
            }
        });

        // Create Ad content callback
        AdCallback contentCallback = new AdCallback() {
            @Override
            public void onShown(@NonNull AdStatusHandler adStatusHandler) {
                Log.d(SampleApplication.TAG,
                        "Interstitial Ad shown from " + adStatusHandler.getNetwork());
            }

            @Override
            public void onShowFailed(@NonNull String s) {
                Log.e(SampleApplication.TAG, "Interstitial Ad show failed: " + s);
                label.setText(s);
            }

            @Override
            public void onClicked() {
                Log.d(SampleApplication.TAG, "Interstitial Ad received Click");
            }

            @Override
            public void onComplete() {
            }

            @Override
            public void onClosed() {
                Log.d(SampleApplication.TAG, "Interstitial Ad received Close");
                label.setText("Closed");
            }
        };

        // Any loading mode, except manual,
        // automatically controls the preparation of ads for impressions.
        if (CAS.settings.getLoadingMode() == LoadingManagerMode.Manual) {
            findViewById(R.id.loadInterBtn).setOnClickListener(v -> {
                label.setText("Loading");
                manager.loadInterstitial();
            });
        } else {
            label.setText("Loading");
            findViewById(R.id.loadInterBtn).setVisibility(View.GONE);
        }

        findViewById(R.id.showInterBtn).setOnClickListener(v -> {
            if (manager.isInterstitialReady())
                manager.showInterstitial(this, contentCallback);
            else
                Log.e(SampleApplication.TAG, "Interstitial Ad not ready to show");
        });
    }

    private void createRewarded(MediationManager manager) {
        TextView label = findViewById(R.id.rewardedStatusText);

        // Set Ad load callback
        manager.getOnAdLoadEvent().add(new AdLoadCallback() {
            @Override
            public void onAdLoaded(@NonNull AdType adType) {
                if (adType == AdType.Rewarded) {
                    label.setText("Loaded");
                    Log.d(SampleApplication.TAG, "Interstitial Ad loaded and ready to show");
                }
            }

            @Override
            public void onAdFailedToLoad(@NonNull AdType adType, @Nullable String s) {
                if (adType == AdType.Rewarded) {
                    label.setText(s);
                    Log.d(SampleApplication.TAG, "Interstitial Ad received error: " + s);
                }
            }
        });

        // Create Ad content callback
        AdCallback contentCallback = new AdCallback() {
            @Override
            public void onShown(@NonNull AdStatusHandler adStatusHandler) {
                Log.d(SampleApplication.TAG,
                        "Rewarded Ad shown from " + adStatusHandler.getNetwork());
            }

            @Override
            public void onShowFailed(@NonNull String s) {
                Log.e(SampleApplication.TAG, "Rewarded Ad show failed: " + s);
                label.setText(s);
            }

            @Override
            public void onClicked() {
                Log.d(SampleApplication.TAG, "Rewarded Ad received Click");
            }

            @Override
            public void onComplete() {
            }

            @Override
            public void onClosed() {
                Log.d(SampleApplication.TAG, "Rewarded Ad received Close");
                label.setText("Closed");
            }
        };

        // Any loading mode, except manual,
        // automatically controls the preparation of ads for impressions.
        if (CAS.settings.getLoadingMode() == LoadingManagerMode.Manual) {
            findViewById(R.id.loadRewardedBtn).setOnClickListener(v -> {
                label.setText("Loading");
                manager.loadRewardedAd();
            });
        } else {
            label.setText("Loading");
            findViewById(R.id.loadRewardedBtn).setVisibility(View.GONE);
        }

        findViewById(R.id.showRewardedBtn).setOnClickListener(v -> {
            if (manager.isRewardedAdReady())
                manager.showRewardedAd(this, contentCallback);
            else
                Log.e(SampleApplication.TAG, "Rewarded Ad not ready to show");
        });
    }

}

File: app/src/main/java/com/cleversolutions/ads/android/sampleapp/SampleActivity.kt
package com.cleversolutions.ads.android.sampleapp

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.cleversolutions.ads.*
import com.cleversolutions.ads.android.CAS
import com.cleversolutions.ads.android.CASBannerView
import kotlinx.android.synthetic.main.activity_main.*

class SampleActivity : AppCompatActivity(), AdLoadCallback {
    companion object {
        const val TAG = "CAS Sample"
    }

    var manager: MediationManager? = null
    var bannerView: CASBannerView? = null
    var statusAdViews: Map<AdType, TextView> = emptyMap()

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        statusAdViews = mapOf(
            AdType.Banner to bannerStatusText,
            AdType.Interstitial to interStatusText,
            AdType.Rewarded to rewardedStatusText
        )

        statusAdViews.forEach {
            it.value.text = "Loading"
        }

        // Validate Integration in log
        CAS.validateIntegration(this, ContentRating.MA)

        // Get current SDK version
        casVersionText.text = "version: " + CAS.getSDKVersion()

        // Set Ads Settings
        CAS.settings.debugMode = true

        // Initialize SDK
        val manager = CAS.initialize(
            this,
            null,
            AdTypeFlags.Everything,
            true
        )
        this.manager = manager
        manager.bannerSize = AdSize.BANNER

        // Subscribe loading ad event
        manager.onAdLoadEvent.add(this)

        // Create Banner View
        val bannerView = CASBannerView(this, manager)
        this.bannerView = bannerView

        bannerView.position = AdPosition.BottomCenter
        bannerView.listener = AdListener(this, AdType.Banner)

        // Attach banner View
        addContentView(
            bannerView,
            ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
            )
        )

        setDefaultSizeBtn.setOnClickListener {
            manager.bannerSize = AdSize.BANNER
        }

        setAdaptiveSizeBtn.setOnClickListener {
            manager.bannerSize = AdSize.getAdaptiveBannerInScreen(this)
        }

        setLedearboardSizeBtn.setOnClickListener {
            manager.bannerSize = AdSize.LEADERBOARD
        }

        setMrecSizeBtn.setOnClickListener {
            manager.bannerSize = AdSize.MEDIUM_RECTANGLE
        }

        showBannerBtn.setOnClickListener {
            bannerView.visibility = View.VISIBLE
        }

        hideBannerBtn.setOnClickListener {
            bannerView.visibility = View.GONE
        }

        showInterBtn.setOnClickListener {
            manager.show(AdType.Interstitial, AdListener(this, AdType.Interstitial))
        }

        showRewardedBtn.setOnClickListener {
            manager.show(AdType.Rewarded, AdListener(this, AdType.Rewarded))
        }
    }

    override fun onAdFailedToLoad(type: AdType, error: String?) {
        runOnUiThread {
            Log.d(TAG, "$type Ad Failed to Load: $error")
            statusAdViews[type]?.text = error
        }
    }

    override fun onAdLoaded(type: AdType) {
        runOnUiThread {
            Log.d(TAG, "$type Ad Loaded")
            statusAdViews[type]?.text = "Ready"
        }
    }


}
